Providers/Profile.swift

```diff
@@ -382,7 +382,10 @@ public class BrowserProfile: Profile {
      * that this is initialized first.
      */
     private lazy var places: protocol<BrowserHistory, Favicons, SyncableHistory, ResettableSyncStorage, HistoryRecommendations> = {
-        return SQLiteHistory(db: self.db, prefs: self.prefs)!
+        guard let history = SQLiteHistory(db: self.db, prefs: self.prefs) else {
```

Comment:
The only reason this can return `nil` is if it fails to initialize tables. Of course, it will also fail to initialize tables if the DB connection is closed.

And of course `SQLiteHistory` already knows how to handle being in its post-init state with a closed `BrowserDB`!

So here's my alternative suggestion:

- Let `SQLiteHistory`'s initializer return non-`nil` if the provided database is closed. Nothing will work, as we expect.
- We don't _need_ to alter schemas via `createOrUpdate` if the DB starts off closed — the schema won't change at runtime, and a closed DB means we had successfully opened by this process before.